controllers/datadogagent/clusteragent_test.go

```diff
@@ -193,6 +193,49 @@ func Test_newClusterAgentDeploymentFromInstance(t *testing.T) {
 				},
 			},
 		},
+		{
+			name:            "defaulted case with DatadogFeature orchestrator Explorer",
+			agentdeployment: test.NewDefaultedDatadogAgent("bar", "foo", &test.NewDatadogAgentOptions{ClusterAgentEnabled: true, OrchestratorExplorerEnabled: true}),
```

Comment:
do we need ClusterAgentEnabled? Ideally orchestrator_explorer.enabled would also enable the cluster agent automatically

Reply:
The above test does not reflect how the controller works.

If the DCA is not set in the config (neither false, nor true) it defaults to true.
So the orchestrator explorer (assuming correctly setup) works in these cases:
- DCA is set to true (1)
- DCA is not setup at all (2)

it does not work in this case:
- DCA is set to false (3)

should we handle case (3) and enforce enabling it? I thought this would be confusing to do so. 
wdyt? @hkaj @clamoriniere

Reply:
I think it is fine like this since the DCA is enabled by default.
If the user disable the DCA on purpose, maybe we can add something in the status of the `DatadogAgent` CRD to inform the user of a misconfiguration.